packages/components/icons/src/custom-icon/custom-icon.tsx

```diff
@@ -0,0 +1,58 @@
+import { type ReactElement, cloneElement } from 'react';
+import {
+  filterAriaAttributes,
+  filterDataAttributes,
+  warning,
+} from '@commercetools-uikit/utils';
+import InlineSvg from '../inline-svg/inline-svg';
+import { getCustomIconStyles } from './custom-icon.styles';
+
+export type TCustomIconProps = {
+  /**
+   * Indicates the size of the component
+   */
+  size?: '10' | '20' | '30' | '40';
+  /**
+   * Indicates whether the component should display a border
+   */
+  hasBorder?: boolean;
+  /**
+   * An <Icon /> component, must pass either an icon prop or an svg prop
+   */
+  icon: ReactElement | string;
+};
+
+const defaultProps: Required<Pick<TCustomIconProps, 'size' | 'hasBorder'>> = {
+  size: '20',
+  hasBorder: true,
+};
+
+const CustomIcon = (props: TCustomIconProps) => {
+  if (!props.icon) {
```

Comment:
I would avoid this check. The `icon` property is the only one required and it's stated in the docs and in the TS types.
For those rare cases where a consumer forgets to pass the icon, I think they will notice right away.

I'm saying because the less code we write, the less code we need to maintain so I believe we should focus on the things that provide the biggest value.

Reply:
good point, it will be removed